dm/pkg/exstorage/utils.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exstorage
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// AdjustPath adjust rawURL, add uniqueId as path suffix.
+func AdjustPath(rawURL string, uniqueID string) (string, error) {
+	if rawURL == "" {
+		return rawURL, nil
+	}
+	u, err := storage.ParseRawURL(rawURL)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	trimPath := strings.TrimRight(u.Path, "/")
+	// avoid duplicate add uniqueID
+	if uniqueID != "" && !strings.HasSuffix(trimPath, uniqueID) {
+		u.Path = trimPath + "." + uniqueID
+		newURL, err := url.QueryUnescape(u.String())
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+		return newURL, nil
+	}
+
+	return rawURL, nil
+}
+
+// isS3Path judges if rawURL is s3 path.
+func IsS3Path(rawURL string) bool {
+	if rawURL == "" {
+		return false
+	}
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return false
+	}
+	if u.Scheme == "s3" {
+		return true
+	}
+	return false
+}
+
+// CreateExternalStore creates ExternalStore.
+func CreateExternalStore(ctx context.Context, path string) (storage.ExternalStorage, error) {
```

Comment:
i found storage also support local backend, so how about rename `ExternalStorage ` to `Storage` like br?

Reply:
yes, it also support local, let me change the name.